### DISCUSSION TITLE
SuperStream Reconnection

### DISCUSSION THREAD

**Discussion by lngvivek** [rabbitmq/rabbitmq-stream-dotnet-client#333] [Q&A] [answered]
I am using dot net stream client (v 1.7.2) for Super Streams. Using
RabbitMQ.Stream.Client.Reliable.Consumer for subscribing to messages. When TCP connection to RabbitMQ breaks (we restart RabbitMQ instance) the subscriber's connection is not coming back. It remains disconnected. We are passing a ReconnectionStrategy which is retuning true for WhenDisconnected(string connectionIdentifier)
Please suggest what is wrong here.

**Comment by Gsantomaggio** [comment]
Hi @lngvivek,
Thank you for reporting the issue. 

Can you give us more information like:

- Do you use a cluster? or Single nodes?
- Do you have some Client log to share?

**Comment by Gsantomaggio** [comment]
@lngvivek I can reproduce the issue. The disconnect handler tries only one time in the case of super-stream consumer.
So it works in case the broker is available but it does not work when the broker is restarted. 
That's a bug

**Reply by lngvivek** [reply]
Thanks @Gsantomaggio for such quick response.
I was using single node and was trying to collect some logs but you already got this reproduced.

**Reply by Gsantomaggio** [reply]
We have pending a couple a tasks for this client. I will work on it as soon as possible.

**Reply by lngvivek** [reply]
@Gsantomaggio is it possible to get this fix in next release/patch? Will appreciate if can get any timeline by when we can get this.

**Reply by Gsantomaggio** [reply]
@lngvivek I will work on it after this PR rabbitmq-stream-dotnet-client/pull/332.  
- I will release the first `beta.1`  at the end of this week  (If I don't find problems) 
- the `beta.2` should contain the fix for this problem. 

I can't promise anything since I have other tasks to do but I hope to release the `beta.2` within 12/14 days.

**Reply by lngvivek** [reply]
Hey @Gsantomaggio, hope you are doing well!
is this fix available now? Kindly let us know.

**Reply by Gsantomaggio** [reply]
Hi @lngvivek,
We had some other priorities, but we are working on it. 
Please follow https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/issues/336. In the next few days, I will start with some PR. 
The aim is to have the client survive on update and cluster restart.

Please have a bit of patience we will be there as soon as possible :)! 

Thank you.

**Comment by Gsantomaggio** [comment]
@lngvivek do you have a chance to test https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/344 ? 
Thank you.

It is still work in progress but any feedback is welcome.

**Reply by lngvivek** [reply]
@Gsantomaggio Thanks, I will try to test this week.

**Comment by Gsantomaggio** [comment]
@lngvivek FYI https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/discussions/347

**Reply by vivek-singh2** [reply, marked as answer]
@Gsantomaggio I was able to test it on 1.8.0 RC2 . Seems working fine.
Thanks for support!

**Comment by Gsantomaggio** [comment]
closed via https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/discussions/333#discussioncomment-8407648
